perceval/utils/matrix.py

```diff
@@ -145,19 +146,35 @@ def random_unitary(n: int, parameters: Optional[Union[np.ndarray,list]] = None)
         :return: a numeric Matrix
         """
         if parameters is not None:
-            assert len(parameters) == 2*n**2, "parameters has not the right size: should be %d, and is %d" % (
-                2*n**2, len(parameters)
-            )
-            a = np.reshape(parameters[:n**2], (n, n))
-            b = np.reshape(parameters[n**2:], (n, n))
-            u = a + 1j * b
+            warnings.warn("use parametrized_unitary(parameters) instead to create a parametrized unitary "
+                          "matrix, version=0.11", DeprecationWarning)
+            return Matrix.parametrized_unitary(n, parameters)
         else:
             u = np.random.randn(n, n) + 1j*np.random.randn(n, n)
+            (q, r) = np.linalg.qr(u)
+            r_diag = np.sign(np.diagonal(np.real(r)))
+            n_u = np.zeros((n, n))
+            np.fill_diagonal(n_u, val=r_diag)
+            return Matrix(np.matmul(q, n_u))
+
+    @staticmethod
+    def parametrized_unitary(n: int, parameters: Union[np.ndarray,list]) -> MatrixN:
+        r"""static method generating a random unitary matrix
```

Comment:
The method description needs to be updated

Reply:
oops